### DISCUSSION TITLE
How do I restrict navigation to other modules if user has unsaved data in dialog?

### DISCUSSION THREAD

**Discussion by DGoertzen** [serenity-is/Serenity#6744] [Help]
I would like to ask the user if they would like to save their data before navigating away from a dialog.  I have found these two related items:
1) https://github.com/orgs/serenity-is/discussions/5856
2) https://github.com/orgs/serenity-is/discussions/5407

I have implemented the code as demonstrated by https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.Northwind/Modules/Customer/CustomerDialog.ts.   This works great if the user clicks the X in the top right corner.  However, the user can still use the navigation on the left hand side and no warning is given.

@minhhungit suggested this: "use permission on navigation and when user do save, update permission table"

I understand what he means, and could implement that.  However. it seems like a "workaround" and I thought there a better solution may exist.  Is there a better solution to restricting navigation away from a dialog than changing the user permissions, or is that what I should implement?

Thanks.

**Comment by minhhungit** [comment]
@DGoertzen is this exact what you want to do do ?
https://github.com/serenity-is/Serenity/issues/814#issuecomment-363510483

**Reply by DGoertzen** [reply]
Hi @minhhungit .  Thanks for taking the time to response!  My understanding is that code link you sent me enables "check with user before saving to see if they really want to save".  However, what I am looking for is "don't let user accidentally navigate away from the page without saving if they have unsaved data that they want to save"  So, I think that is not what I want to do.

**Comment by minhhungit** [comment]
Oh I'm sorry I accident gave you wrong link, you can use `DialogUtils.pendingChangesConfirmation`

Check sample code here https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.Northwind/Modules/Customer/CustomerDialog.ts#L28

The same way that other guys suggested you before :)
